types/vote_set.go

```diff
@@ -539,13 +539,25 @@ func (voteSet *VoteSet) MakeCommit() *Commit {
 	if voteSet.maj23 == nil {
 		cmn.PanicSanity("Cannot MakeCommit() unless a blockhash has +2/3")
 	}
+	blockID := *voteSet.maj23
 
 	// For every validator, get the precommit
 	votesCopy := make([]*Vote, len(voteSet.votes))
 	copy(votesCopy, voteSet.votes)
+	precommits := make([]*CommitSig, len(voteSet.votes))
```

Comment:
Shouldn't we set length to 0 and capacity to `en(voteSet.votes)` and use append if some votes can be nil? Or it's ok to have some `precommits` eq `nil`?

Reply:
It is ok to have some precommits be `nil`. We use the full array so that the `CommitSigs` have matching indices to the validators who signed them. Then when verifying we just skip any that are `nil`. I imagine we could make this work without any `nil` values by using the bit array and counting to determine indices at verification time, but I don't know if that would end up using less total space when serialized.